extensions/notebook/src/book/bookTreeView.ts

```diff
@@ -419,21 +419,33 @@ export class BookTreeViewProvider implements vscode.TreeDataProvider<BookTreeIte
 
 	async findAndExpandParentNode(notebookPath: string): Promise<BookTreeItem> {
 		let bookItem: BookTreeItem = this.currentBook?.getNotebook(notebookPath);
-		// if the node is not expanded getNotebook returns undefined, try to expand the parent node or getChildren of
-		// the root node.
-		if (!bookItem) {
-			// get the parent node and expand it if it's not already
-			let allNodes = this.currentBook?.getAllNotebooks();
-			let book = allNodes ? Array.from(allNodes?.keys())?.filter(x => x.indexOf(notebookPath.substring(0, notebookPath.lastIndexOf(path.sep))) > -1) : undefined;
-			let bookNode = book?.length > 0 ? this.currentBook?.getNotebook(book.find(x => x.substring(0, x.lastIndexOf(path.sep)) === notebookPath.substring(0, notebookPath.lastIndexOf(path.sep)))) : undefined;
-			if (bookNode) {
-				if (this._bookViewer?.visible) {
-					await this._bookViewer.reveal(bookNode, { select: true, focus: false, expand: 3 });
-				} else {
-					await this.getChildren(bookNode);
+		this.currentBook = this.books.length > 1 ? this.books.find(b => notebookPath.indexOf(b.bookPath) > -1) : this.currentBook;
+		// Try to expand the parent node until we get to the notebook
+		if (!bookItem && this.currentBook) {
+			// get the children of root node and expand the nodes to the notebook level.
+			await this.getChildren(this.currentBook.rootNode);
+			// number of levels to expand
+			let depthOfNotebookInBook: number = path.relative(notebookPath, this.currentBook.bookPath)?.split(path.sep)?.length ?? 0;
+			if (this.currentBook.bookItems.length > 0 && depthOfNotebookInBook > 0) {
+				while (depthOfNotebookInBook > 0) {
+					// check if the notebook is available in already expanded levels.
+					bookItem = this.currentBook.bookItems.find(b => b.tooltip === notebookPath);
+					if (bookItem) {
+						break;
+					}
+					// search for the parent book
+					// notebook can be inside the same folder as parent and can be in a different folder as well
+					// so check for both scenarios.
+					let book = this.currentBook.bookItems.find(b => b.tooltip.indexOf(notebookPath.substring(0, notebookPath.lastIndexOf(path.posix.sep))) > -1) ?? this.currentBook.bookItems.find(b => path.relative(notebookPath, b.tooltip)?.split(path.sep)?.length === depthOfNotebookInBook);
```

Comment:
Why do we need to do these two separate checks? Overall it's pretty unclear what these are supposed to do - especially the second one where all you appear to be doing is just looking for a book item that is at the same depth that we're currently looking at? You don't seem to be checking that the paths are actually the same at all.

And are you accounting for the fact that depthOfNotebookInBook will be changing as you through this loop?